be/src/exec/pipeline/runtime_filter_types.h

```diff
@@ -0,0 +1,336 @@
+        _partial_bloom_filter_build_params[idx] = std::move(partial_bloom_filter_build_params);
+        // merge
+        if (1 == _num_active_builders--) {
+            _bloom_filter_descriptors = std::move(bloom_filter_descriptors);
+            merge_in_filters();
+            merge_bloom_filters();
+            return true;
+        }
+        return false;
+    }
+
+    RuntimeInFilters get_total_in_filters() { return _partial_in_filters[0]; }
+
+    RuntimeBloomFilters get_total_bloom_filters() { return _bloom_filter_descriptors; }
+
+    Status merge_in_filters() {
+        bool can_merge_in_filters = true;
+        size_t num_rows = 0;
+        ssize_t k = -1;
+        //squeeze _partial_in_filters and eliminate empty in-filter lists generated by empty hash tables.
+        for (auto i = 0; i < _ht_row_counts.size(); ++i) {
+            auto& in_filters = _partial_in_filters[i];
+            // empty in-filter list is generated by empty hash tables, so skip it.
+            if (_ht_row_counts[i] == 0) {
+                continue;
+            }
+            // empty in-filter list is generated by non-empty hash tables(size>1024), in-filters can not be merged.
+            if (in_filters.empty()) {
+                can_merge_in_filters = false;
+                break;
+            }
+            // move in-filter list indexed by i to slot indexed by k, eliminates holes in the middle.
+            ++k;
+            if (k < i) {
+                _partial_in_filters[k] = std::move(_partial_in_filters[i]);
+            }
+            num_rows += _ht_row_counts[i];
+        }
+
+        can_merge_in_filters = can_merge_in_filters && (num_rows <= 1024) && k >= 0;
+        if (!can_merge_in_filters) {
+            _partial_in_filters[0].clear();
+            return Status::OK();
+        }
+        // only merge k partial in-filter list
+        _partial_in_filters.resize(k + 1);
+
+        auto& total_in_filters = _partial_in_filters[0];
+        for (auto i = 1; i < _partial_in_filters.size(); ++i) {
+            auto& in_filters = _partial_in_filters[i];
+            auto total_in_filter_it = total_in_filters.begin();
+            auto in_filter_it = in_filters.begin();
+            while (total_in_filter_it != total_in_filters.end()) {
+                auto& total_in_filter = *(total_in_filter_it++);
+                auto& in_filter = *(in_filter_it++);
+                if (total_in_filter == nullptr || in_filter == nullptr) {
+                    total_in_filter = nullptr;
+                    continue;
+                }
+                auto* total_in_filter_pred = down_cast<Predicate*>(total_in_filter->root());
+                auto* in_filter_pred = down_cast<Predicate*>(in_filter->root());
+                RETURN_IF_ERROR(total_in_filter_pred->merge(in_filter_pred));
+            }
+        }
+        std::remove(total_in_filters.begin(), total_in_filters.end(), nullptr);
```

Comment:
`std::remove` doesn't modify the original container `total_in_filters`, but returns a new filtered container.
Therefore, it should be `total_in_filters = std::remove(...)`.